hibernate-core/src/main/java/org/hibernate/SessionFactoryObserver.java

```diff
@@ -22,6 +22,14 @@
 	public void sessionFactoryCreated(SessionFactory factory);
 
 	/**
+	 * Callback to indicate that the given factory is about to be closed. This factory is open
+	 * when this method is called.
+	 *
+	 * @param factory The factory closed.
+	 */
+	public void sessionFactoryAboutToClose(SessionFactory factory);
```

Comment:
`sessionFactoryBeforeClose` sounds fine to me and, this way, we maintain a consistency in method naming.